api/peering.go

```diff
@@ -23,53 +35,38 @@ type Peering struct {
 	// Name is the local alias for the peering relationship.
 	Name string
 	// Partition is the local partition connecting to the peer.
-	Partition string `json:"Partition,omitempty"`
+	Partition string `json:",omitempty"`
 	// Meta is a mapping of some string value to any other string value
 	Meta map[string]string `json:",omitempty"`
 	// State is one of the valid PeeringState values to represent the status of
 	// peering relationship.
 	State PeeringState
-	// PeerID is the ID that our peer assigned to this peering.
-	// This ID is to be used when dialing the peer, so that it can know who dialed it.
-	PeerID string
+	// PeerID is the ID that our peer assigned to this peering. This ID is to
+	// be used when dialing the peer, so that it can know who dialed it.
+	PeerID string `json:",omitempty"`
 	// PeerCAPems contains all the CA certificates for the remote peer.
-	PeerCAPems []string
+	PeerCAPems []string `json:",omitempty"`
 	// PeerServerName is the name of the remote server as it relates to TLS.
-	PeerServerName string
+	PeerServerName string `json:",omitempty"`
 	// PeerServerAddresses contains all the connection addresses for the remote peer.
-	PeerServerAddresses []string
+	PeerServerAddresses []string `json:",omitempty"`
 	// CreateIndex is the Raft index at which the Peering was created.
 	CreateIndex uint64
 	// ModifyIndex is the latest Raft index at which the Peering. was modified.
 	ModifyIndex uint64
 }
 
-type PeeringReadRequest struct {
```

Comment:
hm, I recall the rationale for adding these in specifically was to future proof the API. Do we see no reason to keep these around at all?

Reply:
Peering read and Peering list were both http `GET` operations, which do not commonly take payloads, and any request body IIRC behaves slightly differently than something like a POST.

Peering delete is an http `DELETE` operation, which also does not expect a request body.

Having these methods accept a large request struct may give the illusion that you could send the struct over the wire, which you could not. Everything has to be sent via path elements, headers, and query parameters.

In existing endpoints in situations where we wanted extensibility we've done something like adding `PeeringReadOpt{s,ions}` to optionally follow the primary REST identifier (peer name), where each of those gets parked onto the request in a header or parameter. We  could add those if extensibility was a concern.